src/strategies/python.ts

```diff
@@ -100,22 +100,38 @@ export class Python extends BaseStrategy {
       );
     }
 
-    if (!projectName) {
+    const packagePaths = new Set<string>();
+    if (
+      parsedPyProject?.tool?.poetry &&
+      parsedPyProject?.tool?.poetry.packages
+    ) {
+      parsedPyProject?.tool?.poetry.packages.forEach(pkg => {
```

Comment:
nit: the `?.` let's us chain these together. Then in the block, typescript knows that the fields exist so we can drop the `?.`.

```suggestion
      parsedPyProject?.tool?.poetry?.packages
    ) {
      parsedPyProject.tool.poetry.packages.forEach(pkg => {
```